_pages/guides/data-hub-framework.markdown

```diff
@@ -5,17 +5,110 @@ lead_text: ''
+- UI specific code will reside in close proximity to backend specific configurations.  This might not be as preferable to some development teams.
+- Any CRUD operations performed by the Grove project will write to the Data Hub's FINAL database.
+
+
+### How to implement the recommended approach, step-by-step:
+
+Prerequisite: MarkLogic Data Hub has been deployed to your MarkLogic host.  
+
+[Optional] If you previously installed Grove to its default database via the `mlDeploy` command within the `marklogic` directory, remove the Grove-specific configs before proceeding:
+        
+    > cd grove/marklogic
+    > ./gradlew mlUndeploy
+    
+
+### Steps to move Grove modules to the Data Hub
+
+1. Determine the MarkLogic host and port the data-hub-FINAL app server is running on.
+2. At the command line, from within the top level of your Grove project, (the directory with `middle-tier` and `ui` directories inside it), run `grove config` and enter the host and port of the data-hub-FINAL app server.  This will update environment variables for your project.
+3. Create the `ui-modules` directory inside the Data Hub project: `data-hub/src/main/ui-modules`.
+4. Update the Data Hub's `gradle.properties` to register this new location for MarkLogic module deployments.  Add the following line to `gradle.properites`:
+
+    ```
+    mlModulePaths=src/main/ml-modules,src/main/ui-modules
+    ```
+
+5. __Only necessary for Data Hub 5.0.1__ Update the Data Hub's `build.gradle` to use a newer version of ml-gradle.  Add the `dependencies` to the `buildscript` object around line 5: 
+
+    ```JSON
+    dependencies {
+		  classpath "gradle.plugin.com.marklogic:ml-gradle:3.14.0"
+    }
+    ```
+
+6. Copy the contents of the Grove modules to the Data Hub project. Copy 
+    ```
+    grove/marklogic/ml-modules/options
+    grove/marklogic/ml-modules/root 
+    grove/marklogic/ml-modules/services
+    grove/marklogic/ml-modules/transforms
+    
+    to 
+    
+    data-hub/src/main/ui-modules
+    ```
+
+7. Copy the contents of the Grove default user profile and a dictionary document to the Data Hub project. Copy 
+    ```
+    grove/marklogic/ml-content
+    
+    to 
+    
+    data-hub/src/main
+    ```
+
+8. Edit the query options used by your Grove middle-tier's search route (by default, these query options are called `all` and are found in the `data-hub/ui-modules/options/all.xml` file) to remove the following blocks:
+
+    ```xml
+    <constraint name="eyecolor">
+      ...
+    </constraint>
+    <constraint name="docFormat">
+      ...
+    </constraint>
+    <constraint name="gender">
+      ...
+    </constraint>
+    ```
+
+9. Make any other necessary changes to the query options file. For example, you may need to remove the `<additional-query>` specifying that only docs in the `data` collection are returned.  There may be a Data Hub specific collection that is a natural fit to limit the search results for your Grove application.  
+
```

Comment:
Consider adding a note to keep the additional-query in place, or you might start seeing es model defs and such in search results, which is typically not desirable.

Reply:
We could suggest making a change along these lines:

```
      <cts:collection-query>
        <cts:uri>Entity1</cts:uri>
        <cts:uri>Entity2</cts:uri>
        .....
      </cts:collection-query>
```

Reply:
With an added note that this prevents seeing the defs, and flows, etc..

Reply:
Copied your fabulous example in!